site/src/pages/RoadmapPage/AddCoursePopup.tsx

```diff
@@ -1,21 +1,32 @@
-import React, { FC, useState } from 'react';
-import Form from 'react-bootstrap/Form';
+import React, { FC, useEffect, useState } from 'react';
 import Button from 'react-bootstrap/Button';
-import './AddCoursePopup.scss';
-import { useAppDispatch, useAppSelector } from '../../store/hooks';
-import { moveCourse, setShowAddCourse } from '../../store/slices/roadmapSlice';
+import Form from 'react-bootstrap/Form';
 import Modal from 'react-bootstrap/Modal';
 import { quarterDisplayNames } from '../../helpers/planner';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import { moveCourse, setShowAddCourse, setShowSearch } from '../../store/slices/roadmapSlice';
+import './AddCoursePopup.scss';
 
 interface AddCoursePopupProps {}
 
 const AddCoursePopup: FC<AddCoursePopupProps> = () => {
   const dispatch = useAppDispatch();
   const planner = useAppSelector((state) => state.roadmap.yearPlans);
   const showForm = useAppSelector((state) => state.roadmap.showAddCourse);
-  const [year, setYear] = useState(-1);
-  const [quarter, setQuarter] = useState(-1);
+  const currentYearAndQuarter = useAppSelector((state) => state.roadmap.currentYearAndQuarter);
+  const [year, setYear] = useState(
+    currentYearAndQuarter && currentYearAndQuarter.year !== null ? currentYearAndQuarter.year : -1,
+  );
+  const [quarter, setQuarter] = useState(
+    currentYearAndQuarter && currentYearAndQuarter.quarter !== null ? currentYearAndQuarter.quarter : -1,
+  );
```

Comment:
This works but it can be simplified using optional chaining and nullish coalescing.

```suggestion
  const [year, setYear] = useState(currentYearAndQuarter?.year ?? -1);
  const [quarter, setQuarter] = useState(currentYearAndQuarter?.quarter ?? -1);
```